packages/site/cms-site/src/sitePreview/pagesRouter/parsePreviewParams.ts

```diff
@@ -0,0 +1,13 @@
+import { GetStaticPropsContext } from "next";
+
+import { SitePreviewParams } from "../SitePreviewUtils";
+
+function parsePreviewParams(context: GetStaticPropsContext): SitePreviewParams | null {
```

Comment:
I'm not sure about the naming here. Do you have a better idea?

Reply:
Suggestions:

- previewParamsFromContext
- inferPreviewParamsFromContext
- pagesPreviewParams
- legacyPreviewParams
- previewModePreviewParams

Reply:
Difficult. Maybe legacyPagesPreviewParams.

Or, what do you think about moving that into the application?

Reply:
> Or, what do you think about moving that into the application?

Have thought about this as well, since it's essentially a type cast. Might be cumbersome for projects that have multiple different structured content entities. But then one could create a helper in the application

Reply:
> But then one could create a helper in the application

yes, keep the helper function as it is, just move it into the application.

Normally I wouldn't do that - but in this case it's for legacy code - and we have to keep that practically forever in comet.

Reply:
What about sitePreviewApiHandler?

Reply:
It needs pages or legacy in the name, or?

Reply:
I was thinking of moving it to the application code too. However, it's a lot more code then parsePreviewParams

> It needs pages or legacy in the name, or?

Isn't ApiHandler as in NextApiHandler enough?

Reply:
if "ApiHandler" is a term used for pages api only, then yes.

Question if that is obvious for everyone seeing this export.

Reply:
> if "ApiHandler" is a term used for pages api only, then yes.

For app router it's called route handler. Maybe `sitePreviewRoute`, which we export for app router, isn't enough as well?

Reply:
> I'd prefer doing it all in the application

Good idea: 34197ea1042d5389be1cb18b7596b87c85991cd0

Reply:
Currently (matches Next's naming in types):
- App Router: `sitePreviewRoute`
- Pages Router: `sitePreviewApiHandler`

Alternative #1 (matches Next's naming in docs):
- App Router: `sitePreviewRouteHandler`
- Pages Router: `sitePreviewApiRoute`

Alternative #2 (partially matches Next's naming in docs and types):
- App Router: `sitePreviewRouteHandler`
- Pages Router: `sitePreviewApiRouteHandler`

Alternative #3 (shows legacy of pages router):
- App Router: `sitePreviewRoute`
- Pages Router: `legacyPagesRouterSitePreviewApiHandler`

I'm okay with the current naming and #1. #2 would be a compromise, and #3 is too long IMO

Reply:
My vote would be for `#3`.

Pages router is used for our old projects that have a build. And with v7 we consider that legacy, now new projects will be built with that technique, it's only there for easier upgrading. And that's way I vote for "legacy" in exports related to pages router: to avoid people importing it in new projects.

Reply:
Renamed: 43e74bf68d1b6422a02a316952b1abe125c0740d

Should I add a deprecation notice as well?

Reply:
I'd say "legacy" in the same is enough.